packages/block-library/src/query-pagination-next/edit.js

```diff
@@ -1,22 +1,78 @@
 /**
  * WordPress dependencies
  */
-import { __ } from '@wordpress/i18n';
-import { useBlockProps, PlainText } from '@wordpress/block-editor';
+import { __, _x } from '@wordpress/i18n';
+import {
+	useBlockProps,
+	PlainText,
+	InspectorControls,
+} from '@wordpress/block-editor';
+import {
+	PanelBody,
+	__experimentalSegmentedControl as SegmentedControl,
+	__experimentalSegmentedControlOption as SegmentedControlOption,
+} from '@wordpress/components';
 
 export default function QueryPaginationNextEdit( {
-	attributes: { label },
+	attributes: { label, arrow },
 	setAttributes,
 } ) {
 	return (
-		<PlainText
-			__experimentalVersion={ 2 }
-			tagName="a"
-			aria-label={ __( 'Next page link' ) }
-			placeholder={ __( 'Next Page' ) }
-			value={ label }
-			onChange={ ( newLabel ) => setAttributes( { label: newLabel } ) }
-			{ ...useBlockProps() }
-		/>
+		<div className="wp-block-query-pagination-next">
+			<InspectorControls>
+				<PanelBody title={ __( 'Arrow settings' ) }>
+					<SegmentedControl
+						label={ __(
+							'A decorative arrow appended to the next page link.'
+						) }
+						value={ arrow }
+						onChange={ ( value ) => {
+							setAttributes( {
+								arrow: value,
+							} );
+						} }
+						isBlock
+					>
+						<SegmentedControlOption
+							value=""
+							label={ _x(
+								'None',
+								'Arrow option for Query Pagination Next block'
+							) }
+						/>
+						<SegmentedControlOption
+							value="→"
+							label={ _x(
+								'Arrow',
+								'Arrow option for Query Pagination Next block'
+							) }
+						/>
+						<SegmentedControlOption
+							value="»"
```

Comment:
We might need to pick different values (html entities or something similar 🤔 ?) because since `»` rtl behavior is handled by browsers, the css rule we have reverse it again 😄 .